src/osd/OSD.cc

```diff
@@ -5534,6 +5545,12 @@ void OSD::flush_pg_stats()
   osd_lock.Lock();
 }
 
+void OSD::send_beacon(const utime_t& now)
+{
+  dout(20) << __func__ << dendl;
+  last_sent_beacon = now;
+  monc->send_mon_message(new MOSDBeacon());
+}
```

Comment:
Perhaps it should only start doing this when the mons are all luminous or when the require_luminous_osds flag is set.